servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authz;
+
+import com.google.common.collect.ImmutableMap;
+import java.security.AccessControlException;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.projectnessie.cel.tools.ScriptException;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.server.config.QuarkusNessieAuthorizationConfig;
+import org.projectnessie.services.authz.AccessChecker;
+import org.projectnessie.services.authz.AccessContext;
+import org.projectnessie.versioned.NamedRef;
+
+/**
+ * A reference implementation of the {@link AccessChecker} that performs access checks using CEL
+ * expressions.
+ */
+@ApplicationScoped
+public class CelAccessChecker implements AccessChecker {
```

Comment:
Better make this class abstract (+ remove the `@ApplicationScoped` + move it to nessie-services) and have concrete implementations for the individual server implementations. That would give server-specific implementations freedom to extract the principal-name and role/group information.

Reply:
not sure why we would want to do that? The `CelAccessChecker` is a concrete implementation that depends on Quarkus-specific things (like the config) that just shows a way of how it can be done.